GithubRepo/Scenes/Search/ViewModel/SearchViewModel.swift

```diff
@@ -6,50 +6,56 @@
 //
 
 import Foundation
+import RxRelay
+import RxSwift
 
 class SearchViewModel: ViewModelSearching {
     // MARK: - Attributes
 
+    private let disposeBag = DisposeBag()
     private var githubRepository: GithubRepository
-    private(set) var repositoryCellViewModels: [RepositoryCellViewModel]
-    private(set) var loading: Bool
-    private(set) var error: Bool
+    private(set) var repositoryCellViewModels: BehaviorRelay<[RepositoryCellViewModel]>
+    private(set) var state: BehaviorRelay<FetchState>
     private weak var coordinator: SearchCoordinator?
-    weak var delegate: SearchViewModelDelegate?
 
     // MARK: - Constructors
 
     init(coordinator: SearchCoordinator, repository: GithubRepository = GithubMainRepository()) {
         self.coordinator = coordinator
-        self.repositoryCellViewModels = []
-        self.loading = false
-        self.error = false
+        self.repositoryCellViewModels = BehaviorRelay(value: [])
+        self.state = BehaviorRelay(value: .inital)
         self.githubRepository = repository
-        self.githubRepository.delegate = self
+
+        self.bind()
     }
 
     // MARK: - Methods
 
     func fetchRepositories(query: String) {
         githubRepository.fetchRepositories(with: query)
     }
-}
-
-// MARK: - Notifications from Repository
 
-extension SearchViewModel: GithubRepositoryDelegate {
-    func didChangeLoading(loading: Bool) {
-        self.loading = loading
-        self.delegate?.onChangeSearchLoadingState(isLoading: loading)
+    private func bind() {
+        githubRepository.state.subscribe(onNext: {
+            self.onChangeState(state: $0)
+        })
+        .disposed(by: disposeBag)
+        githubRepository.repositories.subscribe(onNext: {
+            self.onChangeRepositories(repos: $0)
+        })
```

Comment:
same as above. Also, this place is a good example of the `map` operator. As you are getting the new values from another observable and mapping it to accept into your relay, you can use the `map` sequence operator.

Also the convention to indent in RxSwift is like this, one operation per line

```suggestion
        githubRepository.repositories
            .map { RepositoryCellViewModel(repository: $0) }
            .drive(self.repositoryCellViewModels)
```